pallets/gear/src/tests.rs

```diff
@@ -13696,6 +13696,57 @@ fn wasm_ref_types_doesnt_work() {
     });
 }
 
+/// Test that the `Gear::run()` extrinsic can only run once per block,
+/// even if somehow included in a block multiple times.
+#[test]
```

Comment:
If gear::run won't be included to the block than QueueReverted event will be deposit when shouldn't

Reply:
It is always included (unless panics during a block creation - in that case `QueueProcessingReverted` event is exactly what is expected), but not more than once in a block. All subsequent attempts (if, let's say, a malicious validator for whatever reason decides to inject another `gear::run()` directly into the block) will result in an error and thereby not be executed.
It's the same type of guards all inherents in substrate have.